File: jpa-separate-ro-rw/src/main/java/com/sight/app/domain/board/comment/Comment.java
package com.sight.app.domain.board.comment;

import com.sight.app.domain.BaseRecordableEntity;
import com.sight.app.domain.board.Board;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Comment extends BaseRecordableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_seq")
    private Long commentSeq;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_seq")
    private Board board;

    private String content;

    @Column(name = "serviceuser_seq")
    private Long writerSeq;

    public Comment(Board board, String content, Long writerSeq) {
        this.board = board;
        this.content = content;
        this.writerSeq = writerSeq;
    }
}

File: JPAwithRecursive/src/main/kotlin/com/sight/blog/app/domain/comment/Comment.kt
package com.sight.blog.app.domain.comment

import com.sight.blog.app.domain.board.Board
import com.sight.blog.common.jpa.BaseRecordableEntity
import javax.persistence.*

@Entity
class Comment(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_seq")
    var seq: Long ?= null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_comment_seq")
    var parent: Comment ?= null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_seq")
    var board: Board,

    var depth: Int,

    var content: String,

): BaseRecordableEntity()